### PR TITLE
StyledShimmer added

### PR DESCRIPTION
**Pull request by shivanshu877**

### CODE REVIEW

File: lib/models/styled_component/styled_shimmer.model.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  factory StyledShimmerConfigModel.fromMap(Map<String, dynamic> map) {
+    return StyledShimmerConfigModel(
+      initialDelay: map['initialDelay'] != null
+          ? int.tryParse(map['initialDelay'].toString())!
+          : 0,
+      repeatDelay: map['repeatDelay'] != null
+          ? int.tryParse(map['repeatDelay'].toString())!
+          : 0,
+      duration: map['duration'] != null
+          ? int.tryParse(map['duration'].toString())!
```
Comment (Suryanshomar7240) [inline lib/models/styled_component/styled_shimmer.model.dart:80]:
```suggestion
          ? int.tryParse(map['duration'].toString())
```

File: lib/models/styled_component/styled_shimmer.model.dart [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+          : 1000,
+      color: map['color'] != null ? map['color'] : null,
+      tiltAngle: map['tiltAngle'] != null
+          ? double.tryParse(map['tiltAngle'].toString())!
```
Comment (Suryanshomar7240) [inline lib/models/styled_component/styled_shimmer.model.dart:84]:
```suggestion
          ? double.tryParse(map['tiltAngle'].toString())
          
          
```

File: lib/ui/components/styled_components/styled_shimmer_widget.dart [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+          milliseconds: styledShimmerConfig != null
+              ? styledShimmerConfig!.duration
+              : 1000),
+      vsync: this,
+    );
+
+    Future.delayed(
+        Duration(
+          milliseconds: styledShimmerConfig != null
+              ? styledShimmerConfig!.initialDelay
```
Comment (Suryanshomar7240) [inline lib/ui/components/styled_components/styled_shimmer_widget.dart:41]: null check sir ?

File: lib/ui/components/styled_components/styled_shimmer_widget.dart [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+              : 2500,
+        ), () {
+      if (mounted) {
+        _controller.forward();
+      }
+
+      _timer = Timer.periodic(
+          Duration(
+            milliseconds: styledShimmerConfig != null
+                ? styledShimmerConfig!.repeatDelay
```
Comment (Suryanshomar7240) [inline lib/ui/components/styled_components/styled_shimmer_widget.dart:51]: null check

**Review by Suryanshomar7240 — CHANGES_REQUESTED** [review]

File: lib/models/styled_component/styled_shimmer.model.dart (hunk 1 above)
Comment (shivanshu877) [reply lib/models/styled_component/styled_shimmer.model.dart:80]: done

File: lib/models/styled_component/styled_shimmer.model.dart (hunk 2 above)
Comment (shivanshu877) [reply lib/models/styled_component/styled_shimmer.model.dart:84]: done

File: lib/ui/components/styled_components/styled_shimmer_widget.dart (hunk 4 above)
Comment (shivanshu877) [reply lib/ui/components/styled_components/styled_shimmer_widget.dart:51]: done

File: lib/ui/components/styled_components/styled_shimmer_widget.dart (hunk 3 above)
Comment (shivanshu877) [reply lib/ui/components/styled_components/styled_shimmer_widget.dart:41]: done

**Review by Suryanshomar7240 — APPROVED** [review]

**Pull request merged**